js/options.js

```diff
@@ -35,14 +35,36 @@
                         view.confirmNumber.bind(view),
                         view.incrementCounter.bind(view)
                     ).then(function() {
+
+                        // helper function to make time-outing promises that
+                        // wait for event and then "clean" after themselves
+                        function timeoutListenerPromise(event) {
+                            return new Promise(function(resolve) {
+                                var timeout;
+                                var listener = function() {
+                                    clearTimeout(timeout);
+                                    resolve();
+                                }
+                                timeout = setTimeout(function() {
+                                    bg.removeEventListener(event, listener);
+                                    resolve();
+                                }, 60000);
+                                bg.addEventListener(event, listener);
+                            });
+                        }
+
```

Comment:
I am not proud at all about this thing. I just didn't want to change the logic too much and didn't want to add deferred (or used the jquery implementation that I don't understand). So I added this weirdness.

It could *maybe* be achieved with Promise.race more effectively, but then I was not sure if there wouldn't be some hanging listener somewhere